core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -396,21 +401,56 @@ protected static void assertValidate(QueryBuilder queryBuilder, int totalExpecte
 
     /**
      * create a random value for either {@link BaseQueryTestCase#BOOLEAN_FIELD_NAME}, {@link BaseQueryTestCase#INT_FIELD_NAME},
-     * {@link BaseQueryTestCase#DOUBLE_FIELD_NAME} or {@link BaseQueryTestCase#STRING_FIELD_NAME}, or a String value by default
+     * {@link BaseQueryTestCase#DOUBLE_FIELD_NAME}, {@link BaseQueryTestCase#STRING_FIELD_NAME} or
+     * {@link BaseQueryTestCase#DATE_FIELD_NAME}, or a String value by default
      */
-    protected static Object randomValueForField(String fieldName) {
-        Object value;
+    protected static Object getRandomValueForFieldName(String fieldName) {
+        Object value = randomAsciiOfLengthBetween(1, 10);
         switch (fieldName) {
-            case BOOLEAN_FIELD_NAME: value = randomBoolean(); break;
-            case INT_FIELD_NAME: value = randomInt(); break;
-            case DOUBLE_FIELD_NAME: value = randomDouble(); break;
-            case STRING_FIELD_NAME: value = randomAsciiOfLengthBetween(1, 10); break;
-            default : value = randomAsciiOfLengthBetween(1, 10);
-        }
+            case STRING_FIELD_NAME:
+                value = rarely() ? randomUnicodeOfLength(10) : value; // unicode in 10% cases
+                break;
+            case INT_FIELD_NAME:
+                value = randomIntBetween(0, 10);
+                break;
+            case DOUBLE_FIELD_NAME:
+                value = randomDouble() * 10;
+                break;
+            case BOOLEAN_FIELD_NAME:
+                value = randomBoolean();
+                break;
+            case DATE_FIELD_NAME:
+                value = new DateTime(System.currentTimeMillis(), DateTimeZone.UTC).toString();
+                break;
+        } // all other fields assigned to random string
         return value;
     }
 
     /**
+     * Helper method to return a mapped or a random field
+     */
+    protected String getRandomFieldName(String... exclude) {
```

Comment:
Really like the unification of the `getRandomValueForFieldName()` helper method above. Here I think it would really improve readability if the method wouldn't take `exclude` arguments but rather all the fieldnames to include in the random choice. More like `randomFrom()` really. I had trouble understanding the tests below, if they read `getRandomFieldNameAndValue(OBJECT_FIELD_NAME)` I first thought I get an Object value back. So even if this makes the list of arguments in the tests longe, I think this should be the other way round.